Modules/_heapqmodule.c

```diff
@@ -59,8 +60,8 @@ siftdown(PyListObject *heap, Py_ssize_t startpos, Py_ssize_t pos)
         arr = _PyList_ITEMS(heap);
         parent = arr[parentpos];
         newitem = arr[pos];
-        arr[parentpos] = newitem;
-        arr[pos] = parent;
+        FT_ATOMIC_STORE_PTR_RELEASE(arr[parentpos], newitem);
```

Comment:
This isn't writing any new object so release memory order isn't really needed, relaxed would work here as Sam suggested on issue as well.